book/tutorials/basic-git.md

```diff
@@ -0,0 +1,201 @@
+
+![](../img/git-repo-structure.png){width=200px}
+
+## Some basic Git jargon
+
+* **Repo** Repository. It is your code and the record of your changes. This record and also the status of your repo is a hidden folder called `.git` . You have a local repo and a remote repo. The remote repo is on GitHub (for in our case) is called `origin`. The local repo is on the JupyterHub.
+* **Stage** Tell Git which changes you want to commit (write to the repo history).
+* **Commit** Write a note about what change the staged files and "commit" that note to the repository record. You are also tagging this state of the repo and you could go back to this state if you wanted.
+* **Push** Push local changes (commits) up to the remote repository on GitHub (`origin`).
+* **Pull** Pull changes on GitHub into the local repository on the JupyterHub.
+* **Git GUIs** A graphical interface for Git (which is command line). Today I will use `jupyterlab-git` which we have installed on JupyterHub.
+* **Shell** A terminal window where we can issue `git` commands.
+
+
+## Overview
+
+Today I will cover the four basic Git/GitHub skills. We will not work with branches today and I won't cover much about merge conflicts. The goal for today is to first get you comfortable with the basic skills and terminology---and also get you set up on the JupyterHub. We will use what is called a  "trunk-based workflow".
+
+### Simple Trunk-based Workflow:
+
+* Make local (on your computer) changes to code.
+* Record what those changes were about and commit to the code change record (history).
+* Push those changes to your remote repository (aka origin)
+
+We'll do this
+
+![](../img/git-linear-flow-2.png)
+
+## Setting up Git
+
+Before we can work with Git in JupyterLab, we need to do some set up.
+
+[First time `git` setup](../preliminary/git)
+
+1. Tell Git who you are and to store your credentials (GitHub login info)
+
+[Show me](https://youtu.be/3CLuOCJMfK0)
+
+2. Get a Personal Access Token from GitHub
+
+Copy the token! You will need it in the next step.
+
+[Show me](https://youtu.be/tkioJhF_gO8)
+
+3. Trigger Git to ask for your password (that personal access token)
+
+You will do this by cloning a private repo. Open a shell and issue this command
+
+```shell
+git clone https://github.com/snowex-hackweek/github_setup_check.github_setup_check.git
+```
+
+It will ask for your GitHub username and password. At the password part, paste in the Personal Access Token.
+
+
+## `jupyterlab-git`
+
+When the instructions say to use or open or click on `jupyterlag-git`, click the icon in the left navbar marked by the red arrow.
+
+![](../img/jupyterlab-git.jpg)
+
+## The Key Skills
+
+* Skill 1: Create a blank repo on GitHub
+* Skill 2: Clone your **GitHub** repo onto the JupyterHub
+* Skill 0: Open your repository in the JupyterLab. 
+* Skill 3: Make some changes and commit those local changes
+* Skill 4: Push the changes to GitHub
+
+* Skill 1b: Fork someone else's GitHub repository
```

Comment:
do you want numerical order here? If you just do: 

* 1 point 
* 1 other point
* 1 next point

the numbers get automatically ordered (https://wordpress.com/support/markdown-quick-reference/)